packages/thyra/core/test/operator_vector/DefaultSpmdMultiVector_UnitTests.cpp

```diff
@@ -101,7 +101,7 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DefaultSpmdMultiVector, defaultConstruct,
   RCP<const MultiVectorBase<Scalar> > mv = createMembers(*vs, g_numCols);
   Teuchos::Array<Scalar> mv_sums(g_numCols);
   Thyra::sums<Scalar>(*mv, mv_sums());
-  out << "sums(mv) = " << mv_sums << "\n";
+  // out << "sums(mv) = " << mv_sums << "\n";
```

Comment:
This is just a badly written test.  You should not even read the elements unless they are initialized.  The test should be something like:

```
{
  const RCP<const VectorSpaceBase<Scalar> > vs = createSpmdVectorSpace<Scalar>(g_localDim);
  const RCP<MultiVectorBase<Scalar> > mv = createMembers(*vs, g_numCols);
  Teuchos::Array<Scalar> mv_sums(g_numCols);

#ifndef THYRA_INITIALIZE_VECS_MULTIVECS_WITH_NANS
  Thyra::assign<Scalar>(mv.ptr(), 1.0);   // Can't touch uninitialized memory!
#endif

  Thyra::sums<Scalar>(*mv, mv_sums());
  out << "sums(mv) = " << mv_sums << "\n";

#ifdef THYRA_INITIALIZE_VECS_MULTIVECS_WITH_NANS
  typedef Teuchos::ScalarTraits<Scalar> ST;
  for (int i = 0; i < g_numCols; ++i) {
    TEST_ASSERT(ST::isnaninf(mv_sums[i]));
  }
#endif
}
```